scenes/cslib_menu.txt

```diff
@@ -149,121 +167,202 @@
 *temp option_slot_id_9 0
 *temp option_slot_id_10 0
 
-*temp ignore 0
-*set ignore ((cur_page - 1) * items_per_page)
-*temp item_count 0
-*temp cur_option 1
-*temp cur_item "@{invert_order ${{prefix&\"_len\"}}|1}"
-*label _filter_loop
-*if (filter = "")
-	*goto _do_not_filter
-*if ({((prefix&"_")&(filter&"_"))&cur_item} = filter_value)
-	*if (filter_type = "NOT_MATCH")
-		*goto _acc
-*elseif (filter_type != "NOT_MATCH")
-	*goto _acc
-*label _do_not_filter
-*set item_count + 1
-*if (ignore < 1) and (cur_option <= items_per_page)
-	*set option_slot_id[cur_option] cur_item
-	*set cur_option + 1
-*if (ignore > 0)
-	*set ignore -1
-*label _acc
-*if (invert_order)
-	*if (cur_item > 1)
-		*set cur_item - 1
-		*goto _filter_loop
-*else
-	*if (cur_item < {prefix&"_count"})
-		*set cur_item + 1
-		*goto _filter_loop
-*if (item_count = 0) and not(allow_cancel)
-	*bug cslib_menu: menu construction failed as there were 0 matching items — consider allowing cancellation
+*temp total_item_count 0
+*temp cur_option_index 1
+*temp cur_item_index "@{const_invert_order ${{prefix&\"_count\"}}|1}"
+*temp items_off_page ((cur_page - 1) * const_items_per_page)
 
-*label _render
-*temp page_count 0
-*if (item_count < items_per_page)
-	*set page_count 1
+*label filter_loop
+*if (filter != "")
+	*gosub _apply_filter {((prefix&"_")&(filter&"_"))&cur_item_index} filter_value
+	*if not(cslib_ret)
```

Comment:
does this mean cslib_ret != ""  ? I really lost track here :-(

Reply:
_apply_filter returns a boolean, so it's checking the result of that.